pkg/linter/linter.go

```diff
@@ -593,6 +598,52 @@ func pkgconfTestLinter(_ context.Context, cfg *config.Configuration, pkgname, pa
 	return fmt.Errorf("pkgconfig directory found")
 }
 
+var isSharedObjectFileRegex = regexp.MustCompile(`\.so$`)
```

Comment:
Is this sufficient to catch everything we want?

I'm genuinely asking here because I'm not familiar with conventions, but I've often seen places where we have version numbers come after `.so` e.g. [abseil](https://apk.dag.dev/https/packages.wolfi.dev/os/x86_64/abseil-cpp-spinlock-wait-20240722.0-r0.apk@sha1:78fca72af0d8b54a16ff2380b2adcf08575c103a/?all=true).

I also wonder if we'd have any "false positives" where a file happens to end in `.so` but isn't a shared object... maybe we can cross that bridge when we get to it.

Reply:
We need a more involved regexp.  Something like:

```
\.so(?:(?:\.[0-9]+)*)*
```

Also, the files ending in `.so` are almost always symlinks to the real shared libraries.  We're interested in running `ldd` against the real objects, IMO (even though `ldd` will dereference the symlinks).

Reply:
We don't need to catch all the symlinks and shared libraries, one is sufficient to identify that the package or subpackage needs testing subsequently I think the expanded regex is good enough.